app/hallebarde/infrastructure/exchange_repository.py

```diff
@@ -1,22 +1,41 @@
-import boto3
+from typing import List, Optional
 
+import boto3
+from boto3.dynamodb.table import TableResource
+from boto3.dynamodb.conditions import Attr
 import hallebarde.config
 from hallebarde.domain.exchange import Exchange
 
 
 def save(exchange: Exchange) -> None:
     table = _get_dynamodb_table()
-    table.put_item(Item={'identifier': exchange.identifier, 'upload_token': exchange.upload_token,
-                         'download_token': exchange.download_token})
+    table.put_item(
+        Item={'identifier': exchange.identifier, 'email': exchange.email, 'upload_token': exchange.upload_token,
+              'download_token': exchange.download_token})
 
 
-def get(identifier: str) -> Exchange:
+def get(identifier: str) -> Optional[Exchange]:
     table = _get_dynamodb_table()
-    response = table.get_item(Key={'identifier': identifier})['Item']
-    return Exchange(response['identifier'], response['upload_token'], response['download_token'])
+    response = table.get_item(Key={'identifier': identifier})
+    if 'Item' in response.keys():
+        return _map_exchange_from_item(response['Item'])
+    else:
+        return None
```

Comment:
I would write it in a more pythonic way, like
`return _map_exchange_from_item(response['Item']) if 'Item' in response.keys() else None`

What do you think ?

Reply:
I have no religion between one-liners pythonic and classical if/else, so I'm ok with that :)